misc/life_mngr/s5_trigger.sh

```diff
@@ -0,0 +1,214 @@
+#! /bin/bash
+
+MAX_ARRAY=6
+
+started="started"
+stopped="stopped"
+SOS="sos"
+UOS="uos"
+SHUTDOWN="shutdown"
+ACKED="acked"
+
+#config
+IP_ADDR="127.0.0.1"
+SOCKET_PORT="8193"
+
+#send message by socket
+send_message()
+{
+	message="$1"
+	echo "sending: $message"
+	echo -ne "$message" >&6 &
+}
+
+#read message by socket
+read_message()
+{
+	read -r -d $'\0' ret_msg <&6
+}
+
+#send message by tty
+send_pre_message()
+{
+	message="$1"
+	echo "sending: $message"
+	echo -ne "$message" > /dev/ttyS1
+}
+
+#read message by tty
+read_pre_message()
+{
+	read -r -d $'\0' ret_msg < /dev/ttyS1
+}
+
+power_off_post_vms() {
+	vm_list=$(acrnctl list)
+	echo $vm_list
+
+	array=($vm_list)
+	num=${#array[@]}
+	echo "vm array size:" $num
+
+	if [ $num -gt $MAX_ARRAY ]; then
+		echo "not normal for acrnctl, or no VMs!"
+		return 0
+	fi
+
+	#shut down post-launched VMs
+	for ((i=0; i<$num; i+=2))
+	do
+		if [ ${array[$i+1]} == $started ]; then
+			echo "shutting down: " ${array[$i]}
+			acrnctl stop ${array[$i]}
+			sleep 5s
+		fi
+	done
+
+	return 1
+}
+
+check_post_vms_status() {
+	#check post-launched VM status for some time
+	check_times=5
+	while [[ $check_times > 0 ]]
+	do
+		vm_list=$(acrnctl list)
+		array=($vm_list)
+		num=${#array[@]}
+		echo "vm status: " $vm_list
+
+		if [ $num -gt $MAX_ARRAY ]; then
+			sleep 5s
+			let check_times--
+			echo "check time:" $check_times
+			continue;
+		fi
+
+		flag=0
+
+		for ((i=0; i<$num; i+=2))
+		do
+			if [ ${array[$i+1]} != $stopped ]; then
+				flag=1
+				break;
+			fi
+		done
+
+		if [ $flag -eq 1 ]; then
+			sleep 5s
+			let check_times--
+			echo "check time:" $check_times
+		else
+			echo "vm status: " $vm_list
+			break;
+		fi
+
+		if [ $check_times -eq 0 ]; then
+			echo "timed out waiting for vms..."
```

Comment:
```suggestion
			echo "Timed out waiting for VMs..."
```

For VMs to what? To shutdown?